tests/test_spec.py

```diff
@@ -0,0 +1,91 @@
+import os
+import numpy as np
+from astropy.io import fits
+from astropy.table import Table
+from corgidrp.data import Dataset, SpectroscopyCentroidPSF
+import corgidrp.spec as steps
+from corgidrp.mocks import create_default_headers
+
+def test_psf_centroid():
+    """
+    Test PSF centroid computation with mock data and validate results
+    """
+    # Define file path to test input
+    file_path = os.path.abspath("tests/test_data/spectroscopy/g0v_vmag6_spc-spec_band3_unocc_CFAM3d_NOSLIT_PRISM3_offset_array.fits")
+    print(f"Attempting to load FITS file: {file_path}")
+
+    # Load FITS data and table (contains true values for testing)
+    with fits.open(file_path) as hdul:
+        psf_array = hdul[0].data
+        psf_table = Table(hdul[1].data)
+        print(f"FITS file loaded. PSF Array shape: {psf_array.shape}")
+        print(f"Metadata rows: {len(psf_table)}")
+
+    # Create Dataset object for traceability
+    dataset = Dataset([file_path])
```

Comment:
we should put each 2-D slice of the PSF array as an individual Image class into the dataset. See: https://github.com/roman-corgi/corgidrp/blob/spectroscopy/tests/test_dataset.py#L52